### ISSUE TITLE
Recipe Collections

### ISSUE THREAD

**Issue by verygoodsoftwarenotvirus** [dinnerdonebetter/dinnerdonebetter#231]
Basically, user-defined collections of recipes

**Comment by verygoodsoftwarenotvirus** [comment]
```sql
CREATE TABLE IF NOT EXISTS recipe_lists (
    id CHAR(27) NOT NULL PRIMARY KEY,
    name text,
    notes TEXT NOT NULL DEFAULT '',
    belongs_to_user CHAR(27) NOT NULL REFERENCES users(id) ON DELETE CASCADE,
    "public" boolean NOT NULL DEFAULT 'false',
    created_at TIMESTAMP WITH TIME ZONE DEFAULT now() NOT NULL,
    last_updated_at TIMESTAMP WITH TIME ZONE,
    archived_at TIMESTAMP WITH TIME ZONE
    UNIQUE (name, belongs_to_user, archived_at)
);

CREATE TABLE IF NOT EXISTS recipe_list_entries (
    id CHAR(27) NOT NULL PRIMARY KEY,
    notes TEXT NOT NULL DEFAULT '',
    recipe_id CHAR(27) NOT NULL REFERENCES recipes(id) ON DELETE CASCADE,
    belongs_to_recipe_collection CHAR(27) NOT NULL REFERENCES recipe_lists(id) ON DELETE CASCADE,
    created_at TIMESTAMP WITH TIME ZONE DEFAULT now() NOT NULL,
    last_updated_at TIMESTAMP WITH TIME ZONE,
    archived_at TIMESTAMP WITH TIME ZONE
    UNIQUE (name, recipe_id, belongs_to_recipe_collection, archived_at)
);
```

**Comment by verygoodsoftwarenotvirus** [comment]
like #296 but for recipes

**Comment by verygoodsoftwarenotvirus** [comment]
closed in favor of #296
